core/src/json_schema.rs

```diff
@@ -0,0 +1,58 @@
+use serde::{Deserialize, Serialize};
```

Comment:
This file is essentially wholesale copied from mongosql-rs. I think there are some simplifications I can make to it but I will do that as the follow-up work right now. Feel free to comment if you have any ideas, but otherwise feel free to skip since this is copy-pasted.